frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -29,50 +28,49 @@ import { DialogService } from '../../../../../../../shared/services/common-dialo
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class FeatureFlagOverviewDetailsSectionCardComponent {
-  @Input() data: FeatureFlag;
-  flagName: string;
-  flagId: string;
-  flagHeaderSubtitle: string;
-  flagCreatedAt: string;
-  flagUpdatedAt: string;
-  flagStatus: FEATURE_FLAG_STATUS;
-  flagOverviewDetails: KeyValueFormat;
+  featureFlag$ = this.featureFlagService.selectedFeatureFlag$;
+  flagOverviewDetails$ = this.featureFlagService.selectedFlagOverviewDetails;
 
   menuButtonItems: IMenuButtonItem[] = [
     { name: 'Edit', disabled: false },
     { name: 'Delete', disabled: false },
   ];
   isSectionCardExpanded = true;
 
-  constructor(private dialogService: DialogService) {}
+  constructor(private dialogService: DialogService, private featureFlagService: FeatureFlagsService) {}
 
-  ngOnInit() {
-    this.flagName = this.data.name;
-    this.flagId = this.data.id;
-    this.flagCreatedAt = this.data.createdAt;
-    this.flagUpdatedAt = this.data.updatedAt;
-    this.flagStatus = this.data.status;
-    this.flagOverviewDetails = {
-      ['Key']: this.data.key,
-      ['Description']: this.data.description,
-      ['App Context']: this.data.context[0],
-      ['Tags']: this.data.tags,
-    };
+  get FEATURE_FLAG_STATUS() {
+    return FEATURE_FLAG_STATUS;
   }
 
   viewLogsClicked(event) {
     console.log('viewLogs Clicked');
     console.log(event);
   }
 
-  onSlideToggleChange(event) {
-    console.log('on Slide Toggle Clicked');
-    console.log(event);
+  onSlideToggleChange(event: MatSlideToggleChange) {
+    const slideToggleEvent = event.source;
+
+    if (slideToggleEvent.checked) {
+      this.openEnableConfirmModel();
+    } else {
+      // this.openDisableConfirmModel();
+      console.log('disable!');
+    }
+
+    // Note: we don't want the toggle to change state immediately because we have to pop a confirmation modal first, so we need override the default and flip it back
+    slideToggleEvent.checked = !slideToggleEvent.checked;
```

Comment:
I tried to disable the toggle, but it isn't working. can you confirm is it right behaviour?